mava/systems/q_learning/anakin/rec_qmix.py

```diff
@@ -0,0 +1,690 @@
+        cfg.network.hidden_state_dim,
+    )
+
+    max_episode_return = -jnp.inf
+    best_params = copy.deepcopy(unreplicate_batch_dim(learner_state.params.online))
+
+    # Main loop:
+    for eval_idx, t in enumerate(
+        range(steps_per_rollout, int(cfg.system.total_timesteps + 1), steps_per_rollout)
+    ):
+        # Learn loop:
+        start_time = time.time()
+        learner_state, (metrics, losses) = update(learner_state)
+        jax.block_until_ready(learner_state)
+
+        # Log:
+        elapsed_time = time.time() - start_time
+        eps = jnp.maximum(
+            cfg.system.eps_min, 1 - (t / cfg.system.eps_decay) * (1 - cfg.system.eps_min)
+        )
+        final_metrics, ep_completed = episode_metrics.get_final_step_metrics(metrics)
+        final_metrics["steps_per_second"] = steps_per_rollout / elapsed_time
+        loss_metrics = losses
+        logger.log({"timestep": t, "epsilon": eps}, t, eval_idx, LogEvent.MISC)
+        if ep_completed:
+            logger.log(final_metrics, t, eval_idx, LogEvent.ACT)
+        logger.log(loss_metrics, t, eval_idx, LogEvent.TRAIN)
+
+        # Evaluate:
+        key, eval_key = jax.random.split(key)
+        eval_keys = jax.random.split(eval_key, cfg.arch.n_devices)
+        eval_params = unreplicate_batch_dim(learner_state.params.online)
+        eval_metrics = evaluator(eval_params, eval_keys, {"hidden_state": eval_hs})
+        jax.block_until_ready(eval_metrics)
+        logger.log(eval_metrics, t, eval_idx, LogEvent.EVAL)
+        episode_return = jnp.mean(eval_metrics["episode_return"])
+
+        # Save best actor params.
+        if cfg.arch.absolute_metric and max_episode_return <= episode_return:
+            best_params = copy.deepcopy(eval_params)
+            max_episode_return = episode_return
+
+        # Checkpoint:
+        if cfg.logger.checkpointing.save_model:
+            # Save checkpoint of learner state
+            unreplicated_learner_state = unreplicate_n_dims(learner_state)
+            checkpointer.save(
+                timestep=t,
+                unreplicated_learner_state=unreplicated_learner_state,
+                episode_return=episode_return,
+            )
+
+    eval_performance = float(jnp.mean(eval_metrics[cfg.env.eval_metric]))
+
+    # Measure absolute metric.
+    if cfg.arch.absolute_metric:
+        eval_keys = jax.random.split(key, cfg.arch.n_devices)
+        eval_batch_size = get_num_eval_envs(cfg, absolute_metric=True)
+        eval_hs = ScannedRNN.initialize_carry(
+            (jax.device_count(), eval_batch_size, cfg.system.num_agents),
+            cfg.network.hidden_state_dim,
+        )
+
+        abs_metric_evaluator = get_eval_fn(eval_env, eval_act_fn, cfg, absolute_metric=True)
+        eval_metrics = abs_metric_evaluator(best_params, eval_keys, {"hidden_state": eval_hs})
+        logger.log(eval_metrics, t, eval_idx, LogEvent.ABSOLUTE)
+
+    logger.stop()
+
+    return float(eval_performance)
```

Comment:
The float is already applied above we can return it as is:
```suggestion
    return eval_performance
```
I guess this comes from the iql 😅